### PR TITLE
Add offline status log

### PR DESCRIPTION
**Pull request by nkdengineer**
### Details

Add offline status log
### Fixed Issues

$ https://github.com/Expensify/App/issues/40250
PROPOSAL: https://github.com/Expensify/App/issues/40250#issuecomment-2072067018

### Tests

1. Open the App
2. Open the log and filter by `offlineStatus`
3. Verify that we have the log about `NetInfo.addEventListener`
4. Go to Settings > Troubleshoot > Force offline
5. Verify that we have the log about force offline
- [x] Verify that no errors appear in the JS console

### Offline tests

None
### QA Steps

1. Open the App
2. Open the log and filter by `offlineStatus`
3. Verify that we have the log about `NetInfo.addEventListener`
4. Go to Settings > Troubleshoot > Force offline
5. Verify that we have the log about force offline
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/161821005/c447622d-c016-4f9f-9601-8df0381af736

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/161821005/8a435841-400c-4c6e-bdb3-1888d63cd8c3

</details>

<details>
<summary>iOS: Native</summary>

Uploading ios-1.mov…

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/161821005/889f29b9-4fc9-4136-b242-41449d31c552

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/161821005/109339fe-e59b-4578-ac1c-13b50fe6c2fe

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/161821005/1f19197f-2430-44b7-90b5-c6128e4fd4ab

</details>

### CODE REVIEW

**Review by Santhosh-Sellavel — APPROVED** [review]
LGTM, tests well!

File: src/libs/NetworkConnection.ts
Comment (hayata-suenaga) [inline src/libs/NetworkConnection.ts:75]:
```suggestion

```

File: src/libs/NetworkConnection.ts
Comment (hayata-suenaga) [inline src/libs/NetworkConnection.ts:74]:
```suggestion
`[NetworkStatus] The force-offline mode was turned off. Getting the device network status from NetInfo. Network state: ${JSON.stringify(state)}. Setting the offline status to: ${Boolean(state.isInternetReachable)}.`,
```

File: src/libs/NetworkConnection.ts
Comment (hayata-suenaga) [inline src/libs/NetworkConnection.ts:70]:
```suggestion
                setOfflineStatus(Boolean(state.isInternetReachable));
```

File: src/libs/NetworkConnection.ts
Comment (hayata-suenaga) [inline src/libs/NetworkConnection.ts:69]: let's declare a constant:

`const isInternetReachable = Boolean(state.isInternetReachable)`

Comment (nkdengineer) [reply src/libs/NetworkConnection.ts:69]: @hayata-suenaga I updated.

**Review by hayata-suenaga — DISMISSED** [review]
thanks for addressing the review 🙇

**Comment by nkdengineer** [comment]
@hayata-suenaga Please help to review again when you have a chance. Thanks.

File: src/libs/NetworkConnection.ts
Comment (hayata-suenaga) [inline src/libs/NetworkConnection.ts:164]:
```suggestion
            `[NetworkStatus] NetInfo.addEventListener event coming, setting "offlineStatus" to ${Boolean(state.isInternetReachable)} with network state: ${JSON.stringify(
```

**Comment by nkdengineer** [comment]
@hayata-suenaga Can we merge the PR now?

**Comment by hayata-suenaga** [comment]
@nkdengineer thank you for asking. we're imposing a merge freeze right now. I'll let you know when we can merge the PR 😄

**Comment by nkdengineer** [comment]
@hayata-suenaga Can we merge this PR now?

**Comment by hayata-suenaga** [comment]
The merge freeze is being gradually being lifted now

I know this is a PR that just adds a logging but it touches the offline status part of the code so I prefer waiting for the complete end to the merge freeze 😄

**Comment by nkdengineer** [comment]
Thanks.

**Comment by mountiny** [comment]
@nkdengineer can you please sync with main?

@hayata-suenaga the freeze is over now, lets get the merge party started 🎉

**Comment by nkdengineer** [comment]
@mountiny I merged main.

**Review by hayata-suenaga — APPROVED** [review]
thank you!

**Comment by pecanoro** [comment]
@nkdengineer This PR caused a deploy blocker here when changing the unreachable condition 😅  https://github.com/Expensify/App/issues/43116

**Pull request merged**
